sdk/client.go

```diff
@@ -533,18 +533,26 @@ func (client *Client) DoActionWithSigner(request requests.AcsRequest, response r
 	// Set whether to ignore certificate validation.
 	// Default InsecureSkipVerify is false.
 	if trans, ok := client.httpClient.Transport.(*http.Transport); ok && trans != nil {
-		trans.TLSClientConfig = &tls.Config{
-			InsecureSkipVerify: client.getHTTPSInsecure(request),
+		if trans.TLSClientConfig != nil {
+			trans.TLSClientConfig.InsecureSkipVerify = client.getHTTPSInsecure(request)
+		} else {
+			trans.TLSClientConfig = &tls.Config{
+				InsecureSkipVerify: client.getHTTPSInsecure(request),
+			}
 		}
 		if proxy != nil && !flag {
 			trans.Proxy = http.ProxyURL(proxy)
 		}
 		client.httpClient.Transport = trans
 	}
 
+	if client.config.Transport != nil {
+		client.httpClient.Transport = client.config.Transport
+	}
```

Comment:
How about move this check to 149 line which finished check of `config.HttpTransport`:

https://github.com/aliyun/alibaba-cloud-sdk-go/blob/f79c69e8dd383537e3e8e2335d7c980dea4cfd66/sdk/client.go#L146-L149

Reply:
fixed